pkg/apis/core/v1beta1/types.go

```diff
@@ -305,19 +305,33 @@ type CDIConfig struct {
 	Status CDIConfigStatus `json:"status,omitempty"`
 }
 
+//Percent is a string that can only be a value between [0,1)
+// XXX: validation doesn't actually seem to do anything.
+// Actually relies on the reconcile not to use bogus values.
+// +kubebuilder:validation:Pattern=`^(0\.[0-9]+|0)$`
+type Percent string
```

Comment:
Why is a string better than using a float?

May be nice to have a `NewPercent` function that validates and returns a `Percent`.

Reply:
An actual float is problematic for portability reasons, and so [the tooling discourages it](https://github.com/kubernetes-sigs/controller-tools/issues/245).